src/action/fetch.js

```diff
@@ -0,0 +1,110 @@
+const MAX_PAGE_ITEMS = 20;
+
+const sendAPI = async (uri) => {
+  const token =
+    "eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIxN2RmMjgwYTBjYjAxMjMwZTk3NjExYmQzMDY4NWJiMiIsInN1YiI6IjY0ZWM3MThlMWZlYWMxMDBlMTZiOTJlMSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.QTHDghdy41TbXdyRHcNGzBuZgEgcIAw_fJHKe7kG3K4";
+  const url = `https://api.themoviedb.org/3/`;
+
+  const options = {
+    method: "GET",
+    headers: {
+      accept: "application/json",
+      Authorization: `Bearer ${token}`,
+    },
+  };
+
+  return fetch(`${url}${uri}`, options);
+};
+
+async function requestManager(func) {
+  let retries = 3;
+  while (retries > 0) {
+    try {
+      return await func();
+    } catch (e) {
+      console.log(e);
+      retries--;
+    }
+  }
+  throw Error("request failed");
+}
+
+const fetchShowAndMovie = async (uri, num) => {
+  const lastpage = Math.ceil(num / MAX_PAGE_ITEMS);
+  const movies = [];
+
+  for (let page = 1; page <= lastpage; page++) {
+    const pageUri = `${uri}&page=${page}`;
+    const response = await sendAPI(pageUri);
+    const data = await response.json();
+
+    const PushMovies = data.results;
+    movies.push(...PushMovies);
+  }
+
+  return movies.slice(0, num);
+};
+
+const fetchWatchID = async (uri) => {
+  const response = await sendAPI(uri);
+  const data = await response.json();
+  return data;
+};
+
+export const getMovieById = async (movieId) => {
+  const uri = `movie/${movieId}?language=en-US`;
+  return requestManager(() => fetchWatchID(uri));
+};
+
+export const getTvShowById = async (tvId) => {
+  const uri = `tv/${tvId}?language=en-US`;
+  return requestManager(() => fetchWatchID(uri));
+};
+
+export const getTopMovies = async (num) => {
+  const uri = `movie/top_rated?language=en-US`;
+  return requestManager(() => fetchShowAndMovie(uri, num));
+};
+
+export const getPopularMovies = async (num) => {
```

Comment:
I do not get the impression that this one has the request manager help.
anyhow, instead of copy pasta the request manager line, put it in the base function:
 'fetchShowandMvoies"